src/main/java/com/puppycrawl/tools/checkstyle/xpath/AbstractNode.java

```diff
@@ -0,0 +1,522 @@
+    }
+
+    /**
+     * Returns string value.
+     * @return string value
+     */
+    @Override
+    public String getStringValue() {
+        throw throwUnsupportedOperationException();
+    }
+
+    /**
+     * Returns CharSequence string value.
+     * @return CharSequence string value
+     */
+    // -@cs[AbbreviationAsWordInName] Implements method from the interface.
+    // No way to change method name, because interface is the part of the Saxon library.
+    @Override
+    public CharSequence getStringValueCS() {
+        throw throwUnsupportedOperationException();
+    }
+
+    /**
+     * Returns fingerprint.
+     * @return fingerprint
+     */
+    @Override
+    public int getFingerprint() {
+        throw throwUnsupportedOperationException();
+    }
+
+    /**
+     * Returns localpart.
+     * @return localpart
+     */
+    @Override
+    public String getLocalPart() {
+        throw throwUnsupportedOperationException();
+    }
+
+    /**
+     * Returns display name.
+     * @return display name
+     */
+    @Override
+    public String getDisplayName() {
+        throw throwUnsupportedOperationException();
+    }
+
+    /**
+     * Returns prefix.
+     * @return prefix
+     */
+    @Override
+    public String getPrefix() {
+        throw throwUnsupportedOperationException();
+    }
+
+    /**
+     * Returns type of the schema.
+     * @return type of the schema
+     */
+    @Override
+    public SchemaType getSchemaType() {
+        throw throwUnsupportedOperationException();
+    }
+
+    /**
+     * Returns AtomicSequence.
+     * @return AtomicSequence
+     * @throws XPathException throws exception
+     */
+    @Override
+    public AtomicSequence atomize() throws XPathException {
+        throw throwUnsupportedOperationException();
+    }
+
+    /**
+     * Returns attribute value.
+     * @param namespace namespace
+     * @param localPart actual name of the attribute
+     * @return attribute value
+     */
+    @Override
+    public String getAttributeValue(String namespace, String localPart) {
+        throw throwUnsupportedOperationException();
+    }
+
+    /**
+     * Generate id method.
+     * @param fastStringBuffer fastStringBuffer
+     */
+    @Override
+    public void generateId(FastStringBuffer fastStringBuffer) {
+        throw throwUnsupportedOperationException();
+    }
+
+    /**
+     * Copy method.
+     * @param receiver receiver
+     * @param index index
+     * @param location location
+     * @throws XPathException throws exception
+     */
+    @Override
+    public void copy(Receiver receiver, int index, Location location) throws XPathException {
+        throw throwUnsupportedOperationException();
+    }
+
+    /**
+     * To be implemented.
+     * @return UnsupportedOperationException exception
+     */
+    protected UnsupportedOperationException throwUnsupportedOperationException() {
+        return new UnsupportedOperationException("Not implemented yet");
+    }
```

Comment:
>  interface with implementations(java8)

We might want to confirm that code coverage will show on these. In regression tool, we saw no coverage on these type of methods and switched to abstract classes.

> "copy-paste" is better than dependency between two classes

I agree with romani.
If atleast one class implements this method but others don't, we should just copy/paste the throw exceptions to the non-abstract classes.
However, if no classes implement the method at all, I think we should just leave it in the `abstract` class and make it `final` so it is clear no one uses it.